app/routes/_layout+/kits.$kitId.manage-assets.tsx

```diff
@@ -0,0 +1,510 @@
+import { FakeCheckbox } from "~/components/forms/fake-checkbox";
+import { ChevronRight } from "~/components/icons/library";
+import Header from "~/components/layout/header";
+import { List } from "~/components/list";
+import { Filters } from "~/components/list/filters";
+import { Button } from "~/components/shared/button";
+import { Image } from "~/components/shared/image";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "~/components/shared/tooltip";
+import { Td } from "~/components/table";
+import { db } from "~/database/db.server";
+import {
+  createBulkKitChangeNotes,
+  createNote,
+  getPaginatedAndFilterableAssets,
+} from "~/modules/asset/service.server";
+import { getUserByID } from "~/modules/user/service.server";
+import { makeShelfError, ShelfError } from "~/utils/error";
+import { isFormProcessing } from "~/utils/form";
+import { data, error, getParams, parseData } from "~/utils/http.server";
+import {
+  PermissionAction,
+  PermissionEntity,
+} from "~/utils/permissions/permission.validator.server";
+import { requirePermission } from "~/utils/roles.server";
+import { tw } from "~/utils/tw";
+
+export async function loader({ context, request, params }: LoaderFunctionArgs) {
+  const authSession = context.getSession();
+  const { userId } = authSession;
+
+  const { kitId } = getParams(params, z.object({ kitId: z.string() }), {
+    additionalData: { userId },
+  });
+
+  try {
+    const { organizationId } = await requirePermission({
+      userId,
+      request,
+      entity: PermissionEntity.kit,
+      action: PermissionAction.update,
+    });
+
+    const [kit, assets] = await Promise.all([
+      db.kit
+        .findFirstOrThrow({
+          where: { id: kitId },
+          select: { id: true, name: true, assets: { select: { id: true } } },
+        })
+        .catch((cause) => {
+          throw new ShelfError({
+            cause,
+            title: "Kit not found!",
+            message:
+              "The kit you are trying to access does not exists or you do not have permission to asset it.",
+            status: 404,
+            label: "Kit",
+          });
+        }),
+      getPaginatedAndFilterableAssets({
+        request,
+        organizationId,
+        excludeSearchFromView: true,
+      }),
+    ]);
+
+    const modelName = {
+      singular: "asset",
+      plural: "assets",
+    };
+
+    return json(
+      data({
+        header: {
+          title: `Manage assets for ${kit.name}`,
+          SubHeading: "Fill up the kit with the assets of your choice.",
+        },
+        searchFieldLabel: "Search assets",
+        searchFieldTooltip: {
+          title: "Search your asset database",
+          text: "Search assets based on asset name or description, category, tag, location, custodian name. Simply separate your keywords by a space: 'Laptop lenovo 2020'.",
+        },
+        kit,
+        ...assets,
+        items: assets.assets.map((asset) => ({
```

Comment:
I think here we will have duplicate data, right? Because we are going to have assets and items contain the same data. Maybe you can remove `assets` before spreading the data?